### ISSUE TITLE
Need to either prevent or handle circular links

### ISSUE THREAD

**Issue by eugene-davis** [IndustrialDragonfly/DEdC#23]
Links returning to their starting object currently breaks things...

**Comment by JoshuaJClark** [comment]
i believe that these need to be handled as it should be a valid option to have a link start and end in the same node

**Comment by eugene-davis** [comment]
I'm not sure, my inclination though is to allow that flexibility. A test will need to be written and logic will need to exist to handle this then.

**Comment by JoshuaJClark** [comment]
this should work by default but will need to be tested

**Comment by eugene-davis** [comment]
Yes, a good test to prevent regression. This feels like something that could go terribly awry in future work...
